drake/core/test/matrix_compare_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "drake/util/eigen_matrix_compare.h"
```

Comment:
This file  should be named eigen_matrix_compare_test.h

https://google.github.io/styleguide/cppguide.html#File_Names